src/v/storage/storage_resources.cc

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "storage_resources.h"
+
+#include "config/configuration.h"
+#include "storage/logger.h"
+#include "vlog.h"
+
+namespace storage {
+
+storage_resources::storage_resources(
+  config::binding<size_t> falloc_step,
+  config::binding<uint64_t> target_replay_bytes,
```

Comment:
falloc_step was a pre-existing size_t parameter (it's strictly an off_t, seastar uses a uint64_t in the interface, but in practice  we never need very huge values).

target_replay_bytes is a uint64_t because it certainly will need more than 32 bits, even if we were on a 32 bit architecture where size_t might be smaller.